test/framework/src/main/java/org/elasticsearch/indices/SystemIndexThreadPoolTests.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices;
+
+import org.elasticsearch.ElasticsearchTimeoutException;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CyclicBarrier;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.startsWith;
+
+/**
+ * Tests to verify that system indices are bypassing user-space thread pools
+ *
+ * <p>We can block thread pools by setting them to one thread and no queue, then submitting
+ * threads that wait on a countdown latch. This lets us verify that operations on system indices
+ * are being directed to other thread pools.</p>
+ *
+ * <p>When implementing this class, don't forget to override {@link ESIntegTestCase#nodePlugins()} if
+ * the relevant system index is defined in a plugin.</p>
+ */
+@ESIntegTestCase.ClusterScope(numDataNodes = 1, numClientNodes = 0)
```

Comment:
Ah this is still not doing any transport-level forking, we need multiple nodes (all with blocked thread pools) for that.

Reply:
Makes sense, though this was useful to me as a checkpoint for getting things working.

I've updated this to block thread pools on any number of nodes, and switched to using a `Phaser` since we can register threads as we go instead of needing to fetch all the thread pool max sizes in advance.